### PR TITLE
Use string unions instead of enums in the SDK

### PR DESCRIPTION
**Pull request by sangaline**
This updates the SDK to use string unions instead of enums for `ProofStatus` and `CircuitStatus`. This allows us to have a single non-type export for the SDK so that we can use the CJS/ESM interop capability of tsup in the build process so that `require().default` isn't required.

```
> require('./dist/lib/index.js')
_class { pollingInterval: 1000 }
```

### CODE REVIEW

File: src/lib/index.ts [hunk 1]
```diff
@@ -2,5 +2,5 @@ import { Client } from "./client";
 
 export default new Client();
 
-export * from "./client";
+export type * from "./client";
```
Comment (sangaline) [inline src/lib/index.ts:5]: This was previously exporting `Client` as a class, this changes it to only export the type of `Client` so we only have a single default export.

File: src/lib/api/services/CircuitsService.ts [hunk 2]
```diff
@@ -112,6 +115,29 @@ export class CircuitsService {
     });
   }
 
+  /**
```
Comment (katiemckeon) [inline src/lib/api/services/CircuitsService.ts:118]: Are these delete methods new or just moved from somewhere else?

**Review by katiemckeon — APPROVED** [review]
LGTM! Was able to use this branch and compile a circuit on prod

File: src/lib/api/services/CircuitsService.ts (hunk 2 above)
Comment (sangaline) [reply src/lib/api/services/CircuitsService.ts:118]: I'm not entirely sure why these weren't being generated before, it must have been some consequence of switching the options on the API client generator. These aren't exposed through the SDK though and we don't use them internally yet either, so shouldn't make much of a difference.

**Pull request merged**